starknet_py/net/full_node_client.py

```diff
@@ -459,16 +468,9 @@ def get_block_identifier(
     return {"block_id": "pending"}
 
 
-def _get_call_payload(tx: Union[InvokeFunction, Call]) -> dict:
-    if isinstance(tx, InvokeFunction):
-        invoke = cast(InvokeFunction, tx)
-        return {
-            "contract_address": convert_to_felt(invoke.contract_address),
-            "entry_point_selector": convert_to_felt(invoke.entry_point_selector),
-            "calldata": [convert_to_felt(i) for i in invoke.calldata],
-        }
+def _get_call_payload(call: Call) -> dict:
     return {
-        "contract_address": convert_to_felt(tx.to_addr),
-        "entry_point_selector": convert_to_felt(tx.selector),
-        "calldata": [convert_to_felt(i) for i in tx.calldata],
+        "contract_address": convert_to_felt(call.to_addr),
+        "entry_point_selector": convert_to_felt(call.selector),
+        "calldata": [convert_to_felt(i) for i in call.calldata],
     }
```

Comment:
If this function is used just in a single place you can create this dictionary there. Same with the `_get_call_payload` in the other client.